src/aggregate/reducers/random_sample.c

```diff
@@ -21,7 +21,7 @@ static void *sampleNewInstance(Reducer *base) {
   size_t blocksize = MAX(10000, sizeof(rsmplCtx) + r->len * sizeof(RSValue *));
   rsmplCtx *ctx = Reducer_BlkAlloc(base, sizeof(*ctx) + r->len * sizeof(RSValue *), blocksize);
   ctx->seen = 0;
-  ctx->samplesArray = RSValue_NewArrayEx(NULL, r->len, 0);
+  ctx->samplesArray = RSValue_NewArray(rm_malloc(sizeof(RSValue *) * r->len), 0);
```

Comment:
why did you change `NULL` to new allocation?

Reply:
To preserve the behavior. Passing the requested length with no values will set the array length to 0 and allocate this amount of memory for future use